gene_finder.py

```diff
@@ -93,11 +163,21 @@ def find_all_ORFs(dna):
         dna: a DNA sequence
         returns: a list of non-nested ORFs
 
+        I did not add any more doctests because this function depends on the
+        previous functions, so assuming the doctests of those were fruitful,
+        one doctest is sufficient for this function. The doctest includes one
+        ORF in each frame, so the function works if the doctest yields true.
+
     >>> find_all_ORFs("ATGCATGAATGTAG")
     ['ATGCATGAATGTAG', 'ATGAATGTAG', 'ATG']
     """
-    # TODO: implement this
-    pass
+    frame1 = find_all_ORFs_oneframe(dna)
+    frame2 = find_all_ORFs_oneframe(dna[1:])
+    frame3 = find_all_ORFs_oneframe(dna[2:])
+
```

Comment:
you could also use a for loop here
`[find_all_ORFs_oneframe(dna[i:]) for i in range 3]`